### PR TITLE
fixes open-url thorium:// and opds:// for mac/linux  and window

### LINKED ISSUE #1614: minor bug: OPDS URL protocol import generates extraneous forward slashes
To reproduce, use an installed build of Thorium, and use your web browser to open the URL `opds://storage.googleapis.com/audiobook_edrlab/feed.json` => Thorium will be offered, chose it and observe the quadruple slashes (see attached screenshot)

Screenshot 2022-01-09 at 20 30 22

### LINKED ISSUE #1736: url protocol handling fails on windows 10 & debian 11
url protocol handling allows to open a file in Thorium from a browser using the thorium:// prefixed url.

url protocol handling not working on windows 10 & debian 11

exemples files : 
1. thorium://streamer.kvmai.com/pub/L3N0cmVhbWVyL21pc2MvZXB1YnMvY2hpbGRyZW5zLWxpdGVyYXR1cmUuZXB1Yg%3D%3D/manifest.json
2. thorium://ngnix-server-ncndadws4q-ey.a.run.app/public/Accessible_EPUB_3.epub
3. https://opds-auth-test-server-aplqpqv3wa-ey.a.run.app/

## On windows

1 and 2 : Catalogs tab view is opened and message is îCannot read properties of indefines (reading 'headers').
image

3 :  opds is displayed by Firefox, Thorium does not open

## On debian

1 and 2 : Thorium opens but nothing happens. No message is displayed.

3 : opds is displayed by Firefox, Thorium does not open

### PR DESCRIPTION
**Pull request by panaC**
It Worked only on mac before but now it should work on 3 platforms : I add it the check of thorium scheme url in start command

fixes #1736 
fixes #1614

### CODE REVIEW

**Comment by panaC** [comment]
Need to test on Windows !

**Comment by panaC** [comment]
opds:// works on windows
thorium:// doesn't works, why ? 
Screenshot 2022-09-08 135046

**Comment by danielweck** [comment]
Thorium doesn't compile on my Windows 11 VM anymore! :) ("out of memory" error ... investigating)

File: package.json [hunk 1]
```diff
@@ -109,13 +109,24 @@
...
-        "thorium",
-        "opds"
-      ]
-    },
+    "protocols": [
+      {
+        "name": "Thorium - publication link",
+        "schemes": [
+          "thorium",
+          "thorium-reader"
```
Comment (panaC) [inline package.json:117]: thorium-reader is not added in the codebase. The extension scheme is checked

File: package.json [hunk 2]
```diff
@@ -109,13 +109,24 @@
...
+        ],
+        "role": "Viewer"
+      },
+      {
+        "name": "Thorium - catalog link",
+        "schemes": [
+          "opds",
+          "feed"
```
Comment (panaC) [inline package.json:125]: feed is not added in the codebase. The extension scheme is checked

**Comment by panaC** [comment]
thanks Daniel for the URL protocol fixes

File: src/main/pdf/extract.ts [hunk 3]
```diff
@@ -46,8 +48,15 @@ export const extractPDFData =
                 // show: false,
                 webPreferences: {
```
Comment (panaC) [inline src/main/pdf/extract.ts:49]: Why added this in this PR ? I think it's better to create a dedicated commit

File: src/package.json [hunk 4]
```diff
@@ -37,7 +37,6 @@
   "homepage": "https://github.com/edrlab/readium-desktop",
   "dependencies": {
     "bindings": "^1.5.0",
-    "mathjax": "^3.2.2",
```
Comment (panaC) [inline src/package.json:40]: same, I think it's better to create a dedicated commit in develop branch for this fixes

**Comment by panaC** [comment]
#1821

File: src/main/pdf/extract.ts (hunk 3 above)
Comment (danielweck) [reply src/main/pdf/extract.ts]: Thank you Pierre https://github.com/edrlab/thorium-reader/pull/1822/files

File: src/package.json (hunk 4 above)
Comment (danielweck) [reply src/package.json]: thank you for the external PR

**Comment by danielweck** [comment]
Tested on Windows, great.

**Pull request merged**